pandas/core/internals/array_manager.py

```diff
@@ -1268,6 +1269,7 @@ def idelete(self, indexer):
 
         self.arrays = [self.arrays[0][to_keep]]
         self._axes = [self._axes[0][to_keep]]
+        return self
```

Comment:
hmm we are not consistent this at all. e.g. iset, insert don't return self

Reply:
we are not, no, but im working on it

Reply:
kk great, yeah would be nice to do this in followns